src/test/java/com/flightstats/hub/config/ClusterServicesRegistrationTest.java

```diff
@@ -0,0 +1,161 @@
+import com.flightstats.hub.dao.aws.S3WriteQueueLifecycle;
+import com.flightstats.hub.metrics.CustomMetricsLifecycle;
+import com.flightstats.hub.metrics.InfluxdbReporterLifecycle;
+import com.flightstats.hub.metrics.PeriodicMetricEmitterLifecycle;
+import com.flightstats.hub.metrics.StatsDReporterLifecycle;
+import com.flightstats.hub.spoke.SpokeTtlEnforcer;
+import com.google.common.util.concurrent.Service;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static com.flightstats.hub.app.HubServices.TYPE.AFTER_HEALTHY_START;
+import static com.flightstats.hub.app.HubServices.TYPE.BEFORE_HEALTH_CHECK;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@Execution(ExecutionMode.SAME_THREAD)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@Slf4j
+class ClusterServicesRegistrationTest {
+
+    @Mock
+    private InfluxdbReporterLifecycle influxdbReporterLifecycle;
+    @Mock
+    private StatsDReporterLifecycle statsDReporterLifecycle;
+    @Mock
+    private DynamoChannelExistenceCheck dynamoChannelExistenceCheck;
+    @Mock
+    private DynamoWebhookExistenceCheck dynamoWebhookExistenceCheck;
+    @Mock
+    private SpokeTtlEnforcer spokeTtlEnforcerRead;
+    @Mock
+    private SpokeTtlEnforcer spokeTtlEnforcerWrite;
+    @Mock
+    private S3WriteQueueLifecycle s3WriteQueueLifecycle;
+    @Mock
+    private CustomMetricsLifecycle customMetricsLifecycle;
+    @Mock
+    private AppProperties appProperties;
+    @Mock
+    private SpokeProperties spokeProperties;
+    @Mock
+    private PeriodicMetricEmitterLifecycle periodicMetricEmitterLifecycle;
+    private ClusterServicesRegistration clusterServicesRegistration;
+
+    @BeforeEach
+    void setup() {
+        clusterServicesRegistration = new ClusterServicesRegistration(
+                influxdbReporterLifecycle,
+                statsDReporterLifecycle,
+                dynamoChannelExistenceCheck,
+                dynamoWebhookExistenceCheck,
+                spokeTtlEnforcerRead,
+                spokeTtlEnforcerWrite,
+                s3WriteQueueLifecycle,
+                customMetricsLifecycle,
+                appProperties,
+                spokeProperties,
+                periodicMetricEmitterLifecycle);
+        HubServices.clear();
+    }
+
+    @Test
+    @Order(1)
```

Comment:
I think I'm being dumb.  It looks like we're resetting all the static state... Why does order matter for these tests?

Reply:
By default, unit tests run concurrently. 
HubServices registration uses static map to store all services registered. Since it is static there is only one instance per class. All unit tests are writing and reading from the same map instance and needs to be executed one after the other.

Reply:
Oh! For some reason I thought that using the `ExecutionMode.SAME_THREAD` would effectively serialize these without needing to add ordering.

Reply:
ExecutionMode.SAME_THREAD serializes tests at Testsuite level. @TestMethodOrder(MethodOrderer.OrderAnnotation.class) does it within Test class. There are a other ways of ordering tests (alphabetically by tests name..).